### PR TITLE
ComsosDiagnosticsContext: Made system usage property bag

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
ComsosDiagnosticsContext: Made system usage property bag

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java [hunk 1]
```diff
@@ -375,14 +376,14 @@ public Set<String> getContactedRegionNames() {
      * a custom {@link CosmosDiagnosticsHandler}
      * @return the system usage
      */
-    public CosmosDiagnosticsSystemUsageSnapshot getSystemUsage() {
+    public Map<String, String> getSystemUsage() {
```
Comment (lmolkova) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java:379]: nit: 
```suggestion
    public Map<String, Object> getSystemUsage() {
```

Object might be a little easier to evolve

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosDiagnosticsSystemUsageSnapshot.java [hunk 2]
```diff
@@ -28,6 +35,12 @@ public final class CosmosDiagnosticsSystemUsageSnapshot {
         this.usedMemory = usedMemory;
         this.availableMemory = availableMemory;
         this.availableProcessors = availableProcessors;
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.<String, String>builder();
+        builder.put("CPU", systemCpuLoad);
+        builder.put("Memory used", usedMemory);
+        builder.put("Memory available", availableMemory);
+        builder.put("Processor count", String.valueOf(availableProcessors));
```
Comment (lmolkova) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosDiagnosticsSystemUsageSnapshot.java:42]: nit: 
use a static/cached  `String.valueOf(availableProcessors)`, or, if you decide to use Object, cache `Integer`

**Review by lmolkova — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosDiagnosticsSystemUsageSnapshot.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosDiagnosticsSystemUsageSnapshot.java:42]: Changed to Map<String, Object>

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java:379]: Changed to Map<String, Object>

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Pull request merged**